documents/coredocument.go

```diff
@@ -695,3 +696,8 @@ func (cd *CoreDocument) IsDIDCollaborator(did identity.DID) (bool, error) {
 func (cd *CoreDocument) GetAccessTokens() ([]*coredocumentpb.AccessToken, error) {
 	return cd.Document.AccessTokens, nil
 }
+
+// SetUsedAnchorRepoAddress sets used anchor repo address.
+func (cd *CoreDocument) SetUsedAnchorRepoAddress(addr common.Address) {
```

Comment:
I thought you recommend generating a new cd when mutating?

Reply:
This will be called when the document is being updated since that is when we know whiic anhor repo is being used. Same as `UpdateLog` which updates author and time of update

Reply:
I find the reasoning for all this, very ambiguous. but ok if you find it easy to deal with.